client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -77,6 +83,103 @@ static Request ping() {
         return new Request("HEAD", "/", Collections.emptyMap(), null);
     }
 
+    static Request bulk(BulkRequest bulkRequest) throws IOException {
+        Params parameters = Params.builder();
+        parameters.withTimeout(bulkRequest.timeout());
+        parameters.withRefreshPolicy(bulkRequest.getRefreshPolicy());
+
+        // Bulk API only supports newline delimited JSON
+        XContentType xContentType = XContentType.JSON;
```

Comment:
it also supports smile. should we validate that the format of all the index/update requests is the same and use that one?  I wouldn't want to make things too complicated though given that most likely nobody uses smile here. People may also theoretically be using cbor or yaml in the single index requests now using the index api, as they don't have to go through REST.